experimental/GraphicalModels/src/PhylogeneticModels.jl

```diff
@@ -0,0 +1,271 @@
+struct PhylogeneticModel
+  n_states::Int
+  graph::Graph{Directed}
+  prob_ring::MPolyRing{QQFieldElem}
+  fourier_ring::MPolyRing{QQFieldElem}
+  trans_matrices::Dict{Edge, MatElem{QQMPolyRingElem}}
+  fourier_params::Dict{Edge, Vector{QQMPolyRingElem}}
+  group::Vector{Vector{Int64}}
+end
+
+struct FourierPhylogeneticModel
+  pm::PhylogeneticModel
+  fourier_ring::MPolyRing{QQFieldElem}
+  fourier_params::Dict{Edge, Vector{QQMPolyRingElem}}
+  group::Vector{Vector{Int64}}
+end
+
+
+# calling elements of the structure
+@doc raw"""
+    graph(pm::PhylogeneticModel)
+
+Return the graph of a PhylogeneticModel `pm`.
+"""
+graph(pm::PhylogeneticModel) = pm.graph
+
+@doc raw"""
+    transition_matrices(pm::PhylogeneticModel)
+
+Returns a dictionary between the edges of the tree specifying a PhylogeneticModel `pm` and attached transition matrices.
+"""
+transition_matrices(pm::PhylogeneticModel) = pm.trans_matrices
+
+@doc raw"""
+    number_states(pm::PhylogeneticModel)
+
+Return the number of states of the PhylogeneticModel `pm`.
+"""
+number_states(pm::PhylogeneticModel) = pm.n_states
+
+@doc raw"""
+    probabilities_ring(pm::PhylogeneticModel)
+
+Returns the ring of probability coordinates of the PhylogeneticModel `pm`.
+"""
+probabilities_ring(pm::PhylogeneticModel) = pm.prob_ring
+
+@doc raw"""
+    fourier_ring(pm::PhylogeneticModel)
+
+    Returns the ring of Fourier coordinates of the PhylogeneticModel `pm`.
+"""
+fourier_ring(pm::PhylogeneticModel) = pm.fourier_ring
+
+@doc raw"""
+    fourier_param(pm::PhylogeneticModel)
+
+Return the Fourier parameters of the PhylogeneticModel `pm` as a vector of eigenvalues of the transition matrices.
+"""
+fourier_parameters(pm::PhylogeneticModel) = pm.fourier_params
+
+group_model(pm::PhylogeneticModel) = pm.group
+
+
+#define group-based models
+@doc raw"""
+    jukes_cantor_model(graph::Graph{Directed})
+
+Creates a `PhylogeneticModel` based on `graph` whose transition matrices are Jukes Cantor matrices. 
+
+# Examples
+```jldoctest
+julia> pm = jukes_cantor_model(graph_from_edges(Directed,[[4,1],[4,2],[4,3]]));
+```
+We can recover the information of `pm`
+```jldoctest
+julia> graph(pm)
+Directed graph with 4 nodes and the following edges:
+(4, 1)(4, 2)(4, 3)
+
+julia> number_states(pm)
+4
+
+julia> prob_ring(pm)
+fourier_ring(pm)
+trans_matrices(pm)
+fourier_params(pm)
+group(pm)
+```
+"""
+function jukes_cantor_model(graph::Graph{Directed})
+    ns = 4
+    ne = n_edges(graph)
+    R, list_a, list_b = polynomial_ring(QQ, :a => 1:ne, :b => 1:ne)
```

Comment:
In general created polynomial rings with `cached=false` (for details please refer to the new caching guidelines I just merged, see PR #3619)
```suggestion
    R, list_a, list_b = polynomial_ring(QQ, :a => 1:ne, :b => 1:ne; cached=false)
```

This also applies to other places in this PR